props/prop_base.h

```diff
@@ -839,6 +839,18 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 
 #endif
 
+#ifdef ENABLE_MENU_DIMBLADE
+   if (change_brightness_){
+        float delta = fmodf(fusor.angle2() - current_tick_angle_, M_PI * 2);//+ or - 2*pi from start point 0
```

Comment:
I think it would make more sense to use the up/down angle to control brightness since it doesn't loop around.
Also, maybe we should just have the brightness and color change be controlled at the same time?

Reply:
From a user perspective this would yield some very akward angles to get the color ánd brightness exactly as you want, when twist is color, and up down is brightness. You might have your color correct, but then your brightness is not what you want. When adjusting your brightness color is most likely affected to.
In my opinion these adjustments should be independent.

What I am aiming for, (and already have working on my sabers) is to have a scroll-able menu, where selection of a menu and adjustment is the same for all menu's you want to add to your saber. This makes selecting and adjusting much more intuitive for an user.

In my saber I have a menu order like this :  Color menu, Volume menu, Brightness menu

What I have (for a 1 button saber);
- longheld when off: menu mode becomes active: ignites saber and selects Color Change menu, adjustment by turning hilt.
- shortclick when in menu mode: next menu: Volume menu, adjustment by twisting hilt only +/- 90 degree twist is needed to go from level 0 to MAX_VOLUME (#define VOLUME controls initial volume after startup/pragramming). Adjustment saturates on max and minlevel, no roll-over.
- shortclick when in menu mode: next menu: Brightness menu, adjusment by twisting hilt, only +/- 90 degree twist is needed to go from 0 to max brightness (saturates on max and minlevel, no roll-over)  
- shortclick when last menu item is active ; exit menu mode. saber remains on.
- longheld while in any menu: exit menu item and menu mode immediately. saber remains on.

Doing it this way, you can add as many custom menu's as you want, it will not require additional buttons for each individual menu, and it is easier to understand how to go trough menu items and adjust them.

For multi-button setups, you can still have certain menu items directly accesable, by some button combinations (like key-shortcuts). It is also up to the installer that wants to make their own propfile, how they would like to acces the menu items.
An example prop file how to use these items should be included, like the saber_fett263_buttons and saber_sa22c_buttons, or this should be added to the default saber.h prop file. 

I am preparing the pull request for the volume menu (with a #define ENABLE_MENU_VOLUME ) as a stand alone addition to the ProffieOS.

Ultimately I would like to have #define ENABLE_MENUS. This will enable ENABLE_MENU_VOLUME and ENABLE_MENU_DIMBLADE (and optinal any other menu that you build yoursels), and at the same time will supress 
 DoChange(EXIT_COLOR_CHANGE);, DoChange(EXIT_DIM_CHANGE); and DoChange(EXIT_VOL_CHANGE);
Otherwise scrolling trough the menus would trigger the exit sound of the previous and the entry sound of the current menu at the same time.
Exiting the last menu item would trigger the  DoChange(EXIT_MENU); and a generic menu exit sound or quote can be used to exit menu mode. This will allow to exit menus with the same common sound (in my case I use a spoken quote 'exiting config mode'.

To clarify this, I will make a youtube video explaining my menu mode, and post it on TRA.